plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -336,12 +305,13 @@ public void primeEditorValue(
             return;
         }
         try {
-            int maxContentLength = DBWorkbench.getPlatform().getPreferenceStore().getInt(ResultSetPreferences.RS_EDIT_MAX_TEXT_SIZE) * 1000;
             if (editor == null) {
                 log.error("Editor is null or undefined");
                 return;
             }
             resetEditorInput();
+            int maxContentLength = valueController.getExecutionContext().getDataSource().getContainer()
```

Comment:
getExecutionContext() has @Nullable annotation. How about to check it?

Reply:
We do the same basically in every other editor, so it was kind of a no-brainer to me to use the same construction